target/linux/qualcommax/ipq807x/base-files/etc/hotplug.d/ieee80211/11_fix_wifi_mac

```diff
@@ -30,4 +30,21 @@ case "$board" in
 		[ "$PHYNBR" = "1" ] && macaddr_add $label_mac 1 > /sys${DEVPATH}/macaddress
 		[ "$PHYNBR" = "2" ] && macaddr_add $label_mac 3 > /sys${DEVPATH}/macaddress
 		;;
-esac
+	netgear,sxr80|\
+	netgear,sxs80)
+		[ "$PHYNBR" = "0" ] && mtd_get_mac_binary boarddata1 0x0c > /sys${DEVPATH}/macaddress
```

Comment:
Why not just set the label MAC and then reference it instead of this whole math?

Reply:
What's the best way of dealing with this? In the stock firmware the device ships with the following layout:
```
LAN MAC: 94:xx:xx:xx:xx:68 (label MAC)
WAN MAC: 94:xx:xx:xx:xx:69
WLAN 2G MAC: 9a:xx:xx:xx:xx:68 (label MAC, but first octet + 6)
WLAN 5G MAC: 94:xx:xx:xx:xx:6a
WLAN 2nd 5G MAC: 94:xx:xx:xx:xx:6b
```
The MACs are stored in boarddata1 but there are only four:
```
0x0 - 94:xx:xx:xx:xx:68 (LAN)
0x6 - 94:xx:xx:xx:xx:69 (WAN)
0xc - 94:xx:xx:xx:xx:6a (First 5G phy0)
0x12 -94:xx:xx:xx:xx:6b (Second 5G phy2)
```
So the MAC for the 2.4G radio isn't set. In the stock firmware they use the label mac but increment the first octet by 6 - which doesn't feel right. 

I could ignore boarddata1 altogether and set them as follows?

First 5G phy0 - label mac + 2
Second 5G phy2 - label mac + 3
2.4G phy1 - label mac + 4

Reply:
OK, I get it now, its rather impossible to get rid of this without finally having NVMEM layout based U-Boot env driver.

Are they still within their OUI prefix if changing first octet?

Reply:
I've got two devices here in different ranges:

34:98:B5
94:A6:7E

So once the first octet is changed the 2.4G radio uses:

3E:98:B5
9A:A6:7E

Which aren't in their prefixes (currently unallocated). I was originally hesitant to even implement this but thought picking another MAC within their prefix might have a higher chance of colliding with another unit if there are a few of them.

Reply:
Seems like a good suggestion to me - 2.4G phy1 could be set to label MAC but with the LA bit set?

Would simplify the fix_mac code to something like:

`macaddr_setbit_la $(get_mac_label)`

Reply:
Yeah, assigning a MAC outside of their prefix is a big no-no, LA is more reasonable

Reply:
I've updated this to set LA and simplify it.